From be3859baf852e2bada6981eba0e97c21acc9628c Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Thu, 19 Oct 2017 07:30:22 +0000
Subject: [PATCH 4/6] Revert "cpufreq: intel_pstate: Drop pointless
 initialization of PID parameters"

This reverts commit ed0dbb75624506e5c4839d43854075e6783d78b5.

Since commit 68e5314e53bd (cpufreq: intel_pstate: Add support for Gemini Lake)
discards upstream changes.

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/cpufreq/intel_pstate.c |   28 +++++++++++++++++++++++++---
 1 files changed, 25 insertions(+), 3 deletions(-)

diff --git a/drivers/cpufreq/intel_pstate.c b/drivers/cpufreq/intel_pstate.c
index 43580e4..2ead181 100644
--- a/drivers/cpufreq/intel_pstate.c
+++ b/drivers/cpufreq/intel_pstate.c
@@ -1049,6 +1049,14 @@ static struct cpu_defaults core_params = {
 };
 
 static const struct cpu_defaults silvermont_params = {
+	.pid_policy = {
+		.sample_rate_ms = 10,
+		.deadband = 0,
+		.setpoint = 60,
+		.p_gain_pct = 14,
+		.d_gain_pct = 0,
+		.i_gain_pct = 4,
+	},
 	.funcs = {
 		.get_max = atom_get_max_pstate,
 		.get_max_physical = atom_get_max_pstate,
@@ -1062,6 +1070,14 @@ static const struct cpu_defaults silvermont_params = {
 };
 
 static const struct cpu_defaults airmont_params = {
+	.pid_policy = {
+		.sample_rate_ms = 10,
+		.deadband = 0,
+		.setpoint = 60,
+		.p_gain_pct = 14,
+		.d_gain_pct = 0,
+		.i_gain_pct = 4,
+	},
 	.funcs = {
 		.get_max = atom_get_max_pstate,
 		.get_max_physical = atom_get_max_pstate,
@@ -1095,6 +1111,14 @@ static const struct cpu_defaults knl_params = {
 };
 
 static const struct cpu_defaults bxt_params = {
+	.pid_policy = {
+		.sample_rate_ms = 10,
+		.deadband = 0,
+		.setpoint = 60,
+		.p_gain_pct = 14,
+		.d_gain_pct = 0,
+		.i_gain_pct = 4,
+	},
 	.funcs = {
 		.get_max = core_get_max_pstate,
 		.get_max_physical = core_get_max_pstate_physical,
@@ -1803,10 +1827,8 @@ static int __init intel_pstate_init(void)
 
 	cpu_def = (struct cpu_defaults *)id->driver_data;
 
+	copy_pid_params(&cpu_def->pid_policy);
 	copy_cpu_funcs(&cpu_def->funcs);
-	
-	if (pstate_funcs.get_target_pstate == get_target_pstate_use_performance)
-		copy_pid_params(&cpu_def->pid_policy);
 
 	if (intel_pstate_msrs_not_valid())
 		return -ENODEV;
-- 
1.7.5.4

